tests/components/cast/test_media_player.py

```diff
@@ -702,8 +702,15 @@ async def test_entity_play_media_cast(hass: HomeAssistant, quick_play_mock):
     chromecast.start_app.assert_called_once_with("abc123")
 
     # Play_media - cast with app name (quick play)
-    await common.async_play_media(hass, "cast", '{"app_name": "youtube"}', entity_id)
-    quick_play_mock.assert_called_once_with(ANY, "youtube", {})
+    await common.async_play_media(
+        hass,
+        "cast",
+        '{"app_name": "youtube", "metadata": {"metadatatype": 3}}',
```

Comment:
This is passing metadata in the `media_id` parameter, not in the `extra` parameter, so you're not testing the new functionality.
tests.components.media_player.common.async_play_media doesn't support it, you can just call the service directly instead:
```py
    await hass.services.async_call(
        media_player.DOMAIN,
        media_player.SERVICE_PLAY_MEDIA,
        {
            ATTR_ENTITY_ID: entity_id,
            media_player.ATTR_MEDIA_CONTENT_TYPE: "cast",
            media_player.ATTR_MEDIA_CONTENT_ID: '{"app_name":"youtube"}',
            media_player.ATTR_MEDIA_EXTRA: {"metadata": {"metadatatype": 3}},
        },
        blocking=True,
    )
```

Reply:
Good point. Changed as suggestion.